python/lsst/obs/base/yamlCamera.py

```diff
@@ -23,181 +23,318 @@
+
+        ix, iy = amp['ixy']
+        perAmpData = amp['perAmpData']
+        if perAmpData:
+            x0, y0 = 0, 0           # origin of data within each amp image
+        else:
+            x0, y0 = ix*xRawExtent, iy*yRawExtent
+
+        rawDataBBox = makeBBoxFromList(amp['rawDataBBox'])  # Photosensitive area
+        xDataExtent, yDataExtent = rawDataBBox.getDimensions()
+        record.setBBox(afwGeom.BoxI(
+            afwGeom.PointI(ix*xDataExtent, iy*yDataExtent), rawDataBBox.getDimensions()))
+
+        rawBBox = makeBBoxFromList(amp['rawBBox'])
+        rawBBox.shift(afwGeom.ExtentI(x0, y0))
+        record.setRawBBox(rawBBox)
+
+        rawDataBBox = makeBBoxFromList(amp['rawDataBBox'])
+        rawDataBBox.shift(afwGeom.ExtentI(x0, y0))
+        record.setRawDataBBox(rawDataBBox)
+
+        rawSerialOverscanBBox = makeBBoxFromList(amp['rawSerialOverscanBBox'])
+        rawSerialOverscanBBox.shift(afwGeom.ExtentI(x0, y0))
+        record.setRawHorizontalOverscanBBox(rawSerialOverscanBBox)
+
+        rawParallelOverscanBBox = makeBBoxFromList(amp['rawParallelOverscanBBox'])
+        rawParallelOverscanBBox.shift(afwGeom.ExtentI(x0, y0))
+        record.setRawVerticalOverscanBBox(rawParallelOverscanBBox)
+
+        rawSerialPrescanBBox = makeBBoxFromList(amp['rawSerialPrescanBBox'])
+        rawSerialPrescanBBox.shift(afwGeom.ExtentI(x0, y0))
+        record.setRawPrescanBBox(rawSerialPrescanBBox)
+
+        if perAmpData:
+            record.setRawXYOffset(afwGeom.Extent2I(ix*xRawExtent, iy*yRawExtent))
+        else:
+            record.setRawXYOffset(afwGeom.Extent2I(0, 0))
+
+        record.setReadoutCorner(readCorners[amp['readCorner']])
+        record.setGain(amp['gain'])
+        record.setReadNoise(amp['readNoise'])
+        record.setSaturation(amp['saturation'])
+        record.setHasRawInfo(True)
+        # flip data when assembling if needs be (e.g. data from the serial at the top of a CCD)
+        flipX, flipY = amp.get("flipXY")
+
+        record.setRawFlipX(flipX)
+        record.setRawFlipY(flipY)
+        # linearity placeholder stuff
+        record.setLinearityCoeffs([float(val) for val in amp['linearityCoeffs']])
+        record.setLinearityType(amp['linearityType'])
+        record.set(linThreshKey, float(amp['linearityThreshold']))
+        record.set(linMaxKey, float(amp['linearityMax']))
+        record.set(linUnitsKey, "DN")
+    return ampCatalog
+
+
+def makeBBoxFromList(ylist):
+    """Given a list [(x0, y0), (xsize, ysize)], probably from a yaml file,
+    return a BoxI
     """
+    (x0, y0), (xsize, ysize) = ylist
+    return afwGeom.BoxI(afwGeom.PointI(x0, y0), afwGeom.ExtentI(xsize, ysize))
+
+
+def makeTransformDict(nativeSys, transformYamlDict, plateScale):
+    """Make a dictionary of TransformPoint2ToPoint2s from yaml, mapping from nativeSys
+
+    Parameters
+    ----------
+    nativeSys : `lsst.afw.cameraGeom.CameraSys`
+    transformYamlDict : `dict`
```

Comment:
Why is YAML important here? At this point the YAML file has been read in and what you have is a python data structure matching the YAML schema.